src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp

```diff
@@ -918,33 +1116,56 @@ void transpose_16NxK(TDST* dst,
                      TDST* tmp,
                      const size_t N,
                      const size_t K,
+                     const size_t block_size,
                      const size_t dst_stride,
                      const size_t src_stride,
-                     const size_t group_size) {
+                     const size_t group_size,
+                     const bool quant_key_bychannel) {
     // The layout for per token per head:
     // |scale(f32)|zeropoint(f32)|quantized feature(u8,idx_1)|quantized feature(u8,idx_2)|...|quantized
     // feature(u8,idx_S)| The quantized feature will start from 8bytes=sizeof(float)+sizeof(float)
     auto s = reinterpret_cast<typename ov::element_type_traits<SRC_PREC>::value_type*>(src);
     auto t = tmp;
     // if group_size not set, the whole row is used as a group
-    for (size_t n = 0; n < N; n++) {
-        size_t src_offset = 0;
-        size_t dst_offset = 0;
-        while (dst_offset < K) {
-            auto f = reinterpret_cast<float*>(s + src_offset);
-            attn_dequant_kernel<TDST, SRC_PREC>(s + src_offset + sizeof(float) * 2,
-                                                t + dst_offset,
-                                                group_size,
-                                                f[0],
-                                                f[1]);
-            src_offset += group_size + sizeof(float) * 2;
-            dst_offset += group_size;
+    if (quant_key_bychannel) {
+        auto p_scales = reinterpret_cast<float*>(s);
+        auto p_zps = p_scales + K;
+        s = s + sizeof(float) * 2 * K;
+        attn_dequant_u8_by_channel_kernel(s, t, N, K, K, src_stride, p_scales, p_zps);
+        for (size_t i = N; i < block_size; i++) {
+            std::memset(t + i * src_stride, 0, sizeof(TDST) * K);
```

Comment:
Should the following `memset` be skipped for quantized case: https://github.com/openvinotoolkit/openvino/blob/7cfc186797a82cb0f9e52c80961fc86f3864ab1f/src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/executor_pa.cpp#L2368-L2380

Reply:
Sure, this could be skipped, previously those codes are not merged into master when I create this PR

Reply:
removed